examples/federated-learning/run.py

```diff
@@ -12,11 +14,44 @@
   # config file was specified
   config_file = sys.argv[1]
   config = tfe.RemoteConfig.load(config_file)
-  tfe.set_config(config)
-  tfe.set_protocol(tfe.protocol.Pond())
-
-session_target = sys.argv[2] if len(sys.argv) > 2 else None
+  config.connect_to_cluster()
+else:
+  config = tfe.EagerLocalConfig([
+      'server0',
+      'server1',
+      'crypto-producer',
+      'model-owner',
+      'data-owner-0',
+      'data-owner-1',
+      'data-owner-2',
+  ])
+
+tfe.set_config(config)
+tfe.set_protocol(tfe.protocol.Pond())
+
+EPOCHS = 1
+BATCH_SIZE = 256
+BATCHES = 60000 // BATCH_SIZE
+
+TRACING = True
+
+@tfe.local_computation
+def build_data_pipeline(validation=False, batch_size=BATCH_SIZE):
```

Comment:
Nope it doesn't build the graph that way. If you put `build_data_pipeline` in a loop the dataset actually gets built each iteration and returns the same data each iteration.

Reply:
that happens even if it's wrapped as a `tf.function`? alternatively you might consider using `tf.device` directly instead of `local_computation`

Reply:
I can double check but I'm pretty sure this makes sense because `tf.function` builds the graph in the same order the code is written and called. It's not doing anything unexpected like how the old graph building code worked its doing exactly what you ask it to.

Reply:
Also good I idea about using tf.device directly. I'll try it out.

Reply:
in the old way we could get a tensor that would contain a new value for each session.run, while only constructing the data graph once. wondered if tf.function could be used for the same here since it will only build the graph once.

Reply:
Yeah maybe you're right. I'll give it a try too.

Reply:
If we think we will be always leaking data because of eager execution, maybe we could say that eager execution is just for simulation/experimentation. For a tutorial, it's going to be a lot easier to experiment with eager. 
cc @jvmancuso

Reply:
You can experiment using `tf.function` the same as you can with eager execution.

Reply:
good point!